include/seqan3/range/view/translation.hpp

```diff
@@ -45,292 +45,657 @@
-     * translation_frames::WITH_FRAME_SHIFTS, translation_frames::SIX_FRAME).
+private:
+    //!\brief The data members of view_translate_single.
+    struct data_members_t
+    {
+        //!\brief The input range (of ranges).
+        urng_t urange;
+        //!\brief The frames that should be used for translation.
+        translation_frames const tf;
+        //!\brief The single frames corresponding to the frames required.
+        std::vector<translation_frames> single_frames{};
+    };
+    //!\brief Storage of data members.
+    std::shared_ptr<data_members_t> data_members;
+
+public:
+    /*!\name Member types
+     * \{
+     */
+    //!\brief The reference_type.
+    using reference         = view_translate_single<urng_t &>;
+    //!\brief The const_reference type.
+    using const_reference   = reference;
+    //!\brief The value_type (which equals the reference_type with any references removed).
+    using value_type        = reference;
+    //!\brief The size_type.
+    using size_type         = size_type_t<urng_t>;
+    //!\brief A signed integer type, usually std::ptrdiff_t.
+    using difference_type   = difference_type_t<urng_t>;
+    //!\brief The iterator type of this view (a random access iterator).
+    using iterator          = detail::random_access_iterator<view_translate const>;
+    //!\brief The const iterator type of this view (same as iterator, because it's a view).
+    using const_iterator    = iterator;
+    //!\}
+
+protected:
+    /*!\name Compatibility
+     * \brief Static constexpr variables that emulate/encapsulate seqan3::compatible_concept (which doesn't work for types during their definition).
+     * \{
      */
-    template <typename irng_t>
     //!\cond
-        requires forward_range_concept<irng_t> &&
-                 nucleotide_concept<std::decay_t<ranges::range_reference_t<std::decay_t<irng_t>>>>
+    // unfortunately we cannot specialise the variable template so we have to add an auxiliary here
+    template <typename t>
+        requires (dimension_v<t> == dimension_v<value_type> + 1) &&
+                 std::is_same_v<remove_cvref_t<innermost_value_type_t<value_type>>,
+                                remove_cvref_t<innermost_value_type_t<t>>>
+    static constexpr bool is_compatible_this_aux = true;
     //!\endcond
-    auto operator()(irng_t && irange, translation_frames const & tf) const
+    //!\}
+
+public:
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    view_translate() = default;
+    constexpr view_translate(view_translate const & rhs) = default;
+    constexpr view_translate(view_translate && rhs) = default;
+    constexpr view_translate & operator=(view_translate const & rhs) = default;
+    constexpr view_translate & operator=(view_translate && rhs) = default;
+    ~view_translate() = default;
+
+    /*!\brief Construct from another range.
+     * \param urange The input range (of ranges).
```

Comment:
Also not a range-of-range... also for consistency I would call it `The underlying (to-be-translated) range.`